src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1267,6 +1356,17 @@ void ReadFromMergeTree::initializePipeline(QueryPipelineBuilder & pipeline, cons
 
     const auto & input_order_info = query_info.getInputOrderInfo();
 
+    /// Construct a proper coordinator
+    if (input_order_info && is_parallel_reading_from_replicas && context->getClientInfo().interface == ClientInfo::Interface::LOCAL)
```

Comment:
Condition `context->getClientInfo().interface == ClientInfo::Interface::LOCAL` is unclear. Please, add a comment.

Reply:
It means that we read from local replica